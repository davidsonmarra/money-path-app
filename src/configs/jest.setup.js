import '@testing-library/react-native/extend-expect';
import {mockGoogleSignIn} from 'src/__mocks__';

jest.useFakeTimers();

const mockTheme = {
  colors: {
    background: '#232E38',
    primary: '#EAC43D',
    secondary: '#2A5C99',
    info: '#14C0CC',
    success: '#A6CD4E',
    title: '#FFF',
    text: '#D9D9D9',
    disabled: '#4F4F4F',
    buttonText: '#FFF',
  },
};

jest.mock('src/hooks/useTheme', () => ({
  useTheme: () => ({
    theme: mockTheme,
    dark: true,
  }),
}));

jest.mock('@react-native-firebase/auth', () => ({
  firebase: {
    auth: () => ({
      signInWithCredential: jest.fn(),
      signOut: jest.fn(),
    }),
  },
  GoogleAuthProvider: {
    credential: jest.fn(),
  },
}));

jest.mock('@react-native-google-signin/google-signin', () => ({
  GoogleSignin: mockGoogleSignIn,
}));

jest.mock('react-native-safe-area-context', () => ({
  ...jest.requireActual('react-native-safe-area-context'),
  useSafeAreaInsets: jest.fn(() => ({top: 0, bottom: 0, left: 0, right: 0})),
}));
