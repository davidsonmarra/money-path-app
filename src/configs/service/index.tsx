import axios from 'axios';

const api = axios.create({
  baseURL: 'https://moneypathapi-production.up.railway.app',
  timeout: 10000,
  headers: {
    Authorization: `eyJhbGciOiJSUzI1NiIsImtpZCI6IjgxYjUyMjFlN2E1ZGUwZTVhZjQ5N2UzNzVhNzRiMDZkODJiYTc4OGIiLCJ0eXAiOiJKV1QifQ.eyJuYW1lIjoiRGF2aWRzb24gTWFycmEgUm9kcmlndWVzIFZpZWlyYSIsInBpY3R1cmUiOiJodHRwczovL2xoMy5nb29nbGV1c2VyY29udGVudC5jb20vYS9BQ2c4b2NJNWtza25jVkozNVJ5aWFxbk1NcmRlWmtkRjFkQ1g3OXlhTHE2dVBORThMMlotMHdNPXM5Ni1jIiwiaXNzIjoiaHR0cHM6Ly9zZWN1cmV0b2tlbi5nb29nbGUuY29tL21vbmV5LXBhdGgtZHAiLCJhdWQiOiJtb25leS1wYXRoLWRwIiwiYXV0aF90aW1lIjoxNzM4MDI2OTg5LCJ1c2VyX2lkIjoiVTFZWm9HRTFWSk5FcTJPOFdpSk83WXhnWW42MyIsInN1YiI6IlUxWVpvR0UxVkpORXEyTzhXaUpPN1l4Z1luNjMiLCJpYXQiOjE3MzgwMjY5ODksImV4cCI6MTczODAzMDU4OSwiZW1haWwiOiJkYXZpZHNvbi52aWVpcmFAc291cGlsYXIuY29tLmJyIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImZpcmViYXNlIjp7ImlkZW50aXRpZXMiOnsiZ29vZ2xlLmNvbSI6WyIxMTQ4OTYzMjk2MzkxMzQ2MzQyNzYiXSwiZW1haWwiOlsiZGF2aWRzb24udmllaXJhQHNvdXBpbGFyLmNvbS5iciJdfSwic2lnbl9pbl9wcm92aWRlciI6Imdvb2dsZS5jb20ifX0.K80hDFbF5FqABuWtlRfcvaSRintNMPiMY6unnRkXo4sCX32OQIL2JAw-7bJPYtvM7QqiMnbbVxVf0YU68EBcJ5ydA8hRChr_Hkitd1nDIDcf8uC4hmuH-03PLXG0VjLGbMLxE9XAdal51oaaiRa0MxT1_4jvuhSNQsQn22YdvBBEiF8-NqfAGCcwDlHoO5sXtYBkQJA-iK_KPAUPcOC2zSF1iLOZChpFhhq1dZKSHX_WsO-QiC0pn38vTgp801mGookgBcZt8FnQCVlol4NWN3sUWL38Vlx73h-Ts2pxU9MqO94OJss19xsZXpOoAAEgMflsfERfvtuOrzVGdYgLTg`,
  },
});

export default api;
