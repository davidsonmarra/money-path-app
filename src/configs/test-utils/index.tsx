import {render} from '@testing-library/react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import AddWalletFormContext from 'src/features/add-wallet/context/add-wallet-form';

interface IAllThemeProviders {
  children: React.ReactElement;
}

const AllTheProviders = ({children}: IAllThemeProviders) => (
  <AddWalletFormContext>
    <SafeAreaView>{children}</SafeAreaView>
  </AddWalletFormContext>
);

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export const renderWithCustomProviders = (component: React.ReactElement) => {
  return {
    render: render(<SafeAreaView>{component}</SafeAreaView>),
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
