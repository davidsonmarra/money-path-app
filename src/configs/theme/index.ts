export {default as colors} from 'src/configs/theme/colors';
