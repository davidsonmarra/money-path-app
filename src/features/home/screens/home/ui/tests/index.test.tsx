import React from 'react';
import {render} from 'src/configs/test-utils';
import HomeContainer from 'src/features/home/screens/home/ui';

const containerInstance = (props: any) => render(<HomeContainer {...props} />);

describe('HomeContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({}).toJSON();
    expect(container).toMatchSnapshot();
  });
});
