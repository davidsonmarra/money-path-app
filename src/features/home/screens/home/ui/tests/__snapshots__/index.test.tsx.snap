// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "off",
        "left": "off",
        "right": "off",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 24,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        Ol√°, 
      </Text>
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Deyverson
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EAC43D",
          "borderRadius": 24,
          "height": 48,
          "justifyContent": "center",
          "width": 48,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "color": "#2A5C99",
            },
          ]
        }
      >
        DY
      </Text>
    </View>
  </RNCSafeAreaView>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "backgroundColor": "#232E38",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "padding": 24,
      }
    }
  />
</RNCSafeAreaView>
`;
